en: refactor get-started (#840)
pingcap/docs-tidb-operator: en/get-started.md

These are the steps this document follows:

1. [Create a Kubernetes test cluster](#create-a-kubernetes-test-cluster)
2. [Deploy TiDB Operator](#deploy-tidb-operator)
3. [Deploy a TiDB cluster and its monitoring services](#deploy-a-tidb-cluster-and-its-monitoring-services)
4. [Connect to a TiDB cluster](#connect-to-tidb)
5. [Upgrade a TiDB cluster](#upgrade-a-tidb-cluster)
6. [Destroy a TiDB cluster](#destroy-a-tidb-cluster)

If you have already created a Kubernetes cluster, you can skip to step 2, [Deploy TiDB Operator](#deploy-tidb-operator).

If you want to do a production-grade deployment, refer to one of these resources:

+ On public cloud:
    - [Deploy TiDB on AWS EKS](deploy-on-aws-eks.md)
    - [Deploy TiDB on GCP GKE (beta)](deploy-on-gcp-gke.md)
    - [Deploy TiDB on Alibaba Cloud ACK](deploy-on-alibaba-cloud.md)

+ In a self-managed Kubernetes cluster:
    - Familiarize yourself with [Prerequisites for TiDB in Kubernetes](prerequisites.md)

>     - [Configure the local PV](configure-storage-class.md#local-pv-configuration) for your Kubernetes cluster to achieve low latency for TiKV

Reviewer: Doesn't this sentence need to be literally consistent with `参考[本地 PV 配置](configure-storage-class.md#本地-pv-配置)让 TiKV 使用高性能本地存储`?

Author: The original EN version was written by Kolbe and then translated to Chinese by another Chinese colleague, so I'm not sure which version is correct.
Let's leave it to @dragonly's discretion.

Reviewer 2: I think "high performance" may be more accurate in this context, because local PV should achieve low latency and high throughput. What do you think @DanielZhangQD

> **Warning:**
>

> > The kind cluster is for test purposes only. **Do not use** in production!

Reviewer: ```suggestion
> The kind cluster is for test purposes only. **Do not use** in production.
```

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

> You're now ready to [deploy TiDB Operator](#deploy-tidb-operator)!

Reviewer: Please be careful about `!` in tech docs.

This section describes how to deploy a Kubernetes cluster using minikube.

[Minikube](https://minikube.sigs.k8s.io/docs/start/) can start a local Kubernetes cluster inside a VM on your laptop. It works on macOS, Linux, and Windows.

> **Warning:**
>

> > The minikube cluster is for demonstration purposes only. **Do not use** in production!

Reviewer: ```suggestion
> The minikube cluster is for demonstration purposes only. **Do not use** it for production!
```

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

> You're now ready to [deploy TiDB Operator](#deploy-tidb-operator)!

Reviewer: ```suggestion
You are now ready to [deploy TiDB Operator](#deploy-tidb-operator).
```

kubectl port-forward -n tidb-cluster svc/basic-tidb 4000 > pf4000.out &
```

This command runs in the background and writes its output to a file called `pf4000.out` so we can continue working in the same shell session.

### Connect to the TiDB service

> **Note:**
>
> To connect to TiDB (< v4.0.7) using a MySQL 8.0 client, if the user account has a password, you must explicitly specify `--default-auth=mysql_native_password`. This is because `mysql_native_password` is [no longer the default plugin](https://dev.mysql.com/doc/refman/8.0/en/upgrading-from-previous-series.html#upgrade-caching-sha2-password).

{{< copyable "shell-regular" >}}

``` shell
mysql -h 127.0.0.1 -P 4000 -u root
```

Expected output:

```
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 76
Server version: 5.7.25-TiDB-v4.0.0 MySQL Community Server (Apache License 2.0)

Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>
```

> After connecting to the cluster, you can execute the following commands to verify some of the functionality available in TiDB. (Some of these require TiDB 4.0; if you've deployed an earlier version, upgrade by consulting the [Upgrade the TiDB cluster](#upgrade-a-tidb-cluster) section).

Author: @dragonly PTAL. Do we need to add this in Chinese?

> After connecting to the cluster, you can execute the following commands to verify some of the functionality available in TiDB. (Some of these require TiDB 4.0; if you've deployed an earlier version, upgrade by consulting the [Upgrade the TiDB cluster](#upgrade-a-tidb-cluster) section).

Reviewer 2: I have consulted TiDB developers and confirmed that these 4 commands are all supported before TiDB 4.0. Maybe we should make sure if the statements here are correct?

> After connecting to the cluster, you can execute the following commands to verify some of the functionality available in TiDB. (Some of these require TiDB 4.0; if you've deployed an earlier version, upgrade by consulting the [Upgrade the TiDB cluster](#upgrade-a-tidb-cluster) section).

Author: @dragonly We can safely remove the statement about 4.0 if you see fit.

> After connecting to the cluster, you can execute the following commands to verify some of the functionality available in TiDB. (Some of these require TiDB 4.0; if you've deployed an earlier version, upgrade by consulting the [Upgrade the TiDB cluster](#upgrade-a-tidb-cluster) section).

Reviewer 2: I have tested with TiDB v3.0.10, actually the last SQL is not supported:
```sql
select * from information_schema.cluster_info\G
```

We should only point out this one.

> After connecting to the cluster, you can execute the following commands to verify some of the functionality available in TiDB. (Some of these require TiDB 4.0; if you've deployed an earlier version, upgrade by consulting the [Upgrade the TiDB cluster](#upgrade-a-tidb-cluster) section).

Author: @dragonly @TomShawn Thanks. I updated both en and zh in https://github.com/pingcap/docs-tidb-operator/pull/840/commits/f3eef121f80f16e3b623307268f8217ee0d85982. PTAL again.

The default username and password in Grafana are both `admin`.

For more information about monitoring the TiDB cluster in TiDB Operator, refer to [Monitor a TiDB Cluster Using TidbMonitor](monitor-using-tidbmonitor.md).

## Upgrade a TiDB cluster

TiDB Operator also makes it easy to perform a rolling upgrade of the TiDB cluster. This section describes how to upgrade your TiDB cluster to the "nightly" release.

Before that, first you need to get familiar with two `kubectl` sub-commands:

- `kubectl edit` opens a resource specification in an interactive text editor, where an administrator can make changes and save them. If the changes are valid, they'll be propagated to the cluster resources; if they're invalid, they'll be rejected with an error message. Note that not all elements of the specification are validated at this time; it's possible to save changes that may not be applied to the cluster even though they're accepted.

- `kubectl patch` applies a specification change directly to the running cluster resources. There are several different patch strategies, each of which has various capabilities, limitations, and allowed formats.

### Modify the TiDB cluster version

> In this case, we can use a JSON merge patch to update the version of the TiDB cluster to "nightly":

Author: Added in Chinese.

{{< copyable "shell-regular" >}}

```shell
kubectl patch tc basic -n tidb-cluster --type merge -p '{"spec": {"version": "release-4.0-nightly"} }'
```

Expected output:

```
tidbcluster.pingcap.com/basic patched
```

### Wait for Pods to restart

To follow the progress of the cluster as its components are upgraded, execute the following command. You should see some Pods transition to "Terminating" and then back to "ContainerCreating" and back to "Running". Note the value in the "AGE" pod column to see which pods have restarted.

{{< copyable "shell-regular" >}}

```
watch kubectl get po -n tidb-cluster
```

Expected output:

```
NAME                              READY   STATUS        RESTARTS   AGE
basic-discovery-6bb656bfd-7lbhx   1/1     Running       0          24m
basic-pd-0                        1/1     Terminating   0          5m31s
basic-tidb-0                      2/2     Running       0          2m19s
basic-tikv-0                      1/1     Running       0          4m13s
```

### Forward the TiDB service port

After all Pods have been restarted, you should be able to see that the version number of the cluster has changed.

> Note that any port forwarding you set up in a previous step will need to be re-done, because the pod(s) they forwarded to will have been destroyed and re-created. If the `kubectl port-forward` process is still running in your shell, kill it before forwarding the port again.

Author: Updated the Chinese
